src/cargo/util/rustc.rs

```diff
@@ -42,11 +44,28 @@ impl Rustc {
             triple.to_string()
         };
 
+        let mut cmd = util::process(&path);
```

Comment:
I'm ok with a compromise, yeah. It's likely to be quite rare that Cargo runs against a compiler that doesn't support `--crate-type metadata` due to how our releases work, so having a subar for such a rare situation seems fine.